bin/ch/Debugger.cpp

```diff
@@ -316,13 +316,13 @@ bool Debugger::SetBaseline()
 
     if(file != nullptr)
     {
-        int fileSize = _filelength(_fileno(file));
-        if (fileSize <= MAX_BASELINE_SIZE)
+        long fileSize = _filelength(_fileno(file));
+        if (0 <= fileSize && fileSize <= MAX_BASELINE_SIZE)
         {
-            script = new char[fileSize + 1];
+            script = new char[(size_t)fileSize + 1];
```

Comment:
Is this required?

Reply:
Array size declarations are size_t, and the analysis gets unhappy if we have a different type used in the addition here before the implicit conversion. This is notable for types shorter than size_t, as the overflow would occur before the conversion to size_t. I believe that on all currently-supported platforms this is indeed not required.